clients/python-wrapper/lakefs/reference.py

```diff
@@ -134,24 +132,23 @@ def diff(self,
                                      **kwargs):
             yield Change(**diff.dict())
 
-    def merge_into(self, destination_branch_id: str | Reference, **kwargs) -> str:
+    def merge_into(self, destination_branch: ReferenceType, **kwargs) -> str:
         """
         Merge this reference into destination branch
 
-        :param destination_branch_id: The ID of the merge destination
+        :param destination_branch: The merge destination (either ID or branch object)
```

Comment:
Also here it could get reference ID , branch object and reference object

Reply:
destination branch can only be the branch id or the branch object - it shouldn't be any other reference type. The only reason we use ReferenceType here is because we can import Branch class due to import cycle